src/plugins/discover/public/services/doc_views/components/doc_viewer_table/table.tsx

```diff
@@ -7,67 +7,137 @@
  */
 
 import './table.scss';
-import React, { useCallback, useMemo } from 'react';
-import { EuiInMemoryTable } from '@elastic/eui';
-import { IndexPattern, IndexPatternField } from '../../../../../../data/public';
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFieldSearch,
+  EuiSpacer,
+  EuiTable,
+  EuiTableBody,
+  EuiTableRowCell,
+  EuiTableRow,
+  EuiTableHeader,
+  EuiTableHeaderCell,
+  EuiText,
+  EuiTablePagination,
+  EuiSelectableMessage,
+  EuiI18n,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { debounce } from 'lodash';
+import { Storage } from '../../../../../../kibana_utils/public';
+import { usePager } from '../../../../utils/use_pager';
+import { FieldName } from '../../../../components/field_name/field_name';
 import { flattenHit } from '../../../../../../data/common';
 import { SHOW_MULTIFIELDS } from '../../../../../common';
 import { getServices } from '../../../../kibana_services';
-import { DocViewFilterFn, DocViewRenderProps } from '../../doc_views_types';
-import { ACTIONS_COLUMN, MAIN_COLUMNS } from './table_columns';
+import { DocViewRenderProps, FieldRecordLegacy } from '../../doc_views_types';
 import { getFieldsToShow } from '../../../../utils/get_fields_to_show';
-import { getIgnoredReason, IgnoredReason } from '../../../../utils/get_ignored_reason';
+import { getIgnoredReason } from '../../../../utils/get_ignored_reason';
 import { formatFieldValue } from '../../../../utils/format_value';
 import { isNestedFieldParent } from '../../../../application/main/utils/nested_fields';
-import { ElasticSearchHit } from '../../../../types';
-
-export interface DocViewerTableProps {
-  columns?: string[];
-  filter?: DocViewFilterFn;
-  hit: ElasticSearchHit;
-  indexPattern: IndexPattern;
-  onAddColumn?: (columnName: string) => void;
-  onRemoveColumn?: (columnName: string) => void;
-}
+import { TableFieldValue } from './table_cell_value';
+import { TableActions } from './table_cell_actions';
 
 export interface FieldRecord {
-  action: {
-    isActive: boolean;
-    onFilter?: DocViewFilterFn;
-    onToggleColumn: (field: string) => void;
-    flattenedField: unknown;
-  };
+  action: Omit<FieldRecordLegacy['action'], 'isActive'>;
   field: {
-    displayName: string;
-    field: string;
-    scripted: boolean;
-    fieldType?: string;
-    fieldMapping?: IndexPatternField;
-  };
-  value: {
-    formattedValue: string;
-    ignored?: IgnoredReason;
-  };
+    pinned: boolean;
+    onTogglePinned: (field: string) => void;
+  } & FieldRecordLegacy['field'];
+  value: FieldRecordLegacy['value'];
+}
+
+interface ItemsEntry {
+  pinnedItems: FieldRecord[];
+  restItems: FieldRecord[];
+}
+
+interface BrowseFieldsState {
+  searchField?: string;
+  pinnedFields?: string[];
+  pageSize?: number;
```

Comment:
For now user has to change `pageSize` and `searchField` per each dataView. Would it make sense to save search and page size independently from dataView or user will expect different settings for each of dataView? WDYT @kertal @majagrubic

Reply:
From my perspective it makes sense to store `pageSize` independently. I wonder if there's a case where it makes sense to store `searchField` by dataView ... there might be ... but generally, e.g. if i've got multiple similar filebeat dataViews I'd prefer to keep my search between them.... so I'd think we should also store this value independently, wdyt @majagrubic ?

Reply:
I think it makes sense to store it independently, shouldn't be tied to a dataView.